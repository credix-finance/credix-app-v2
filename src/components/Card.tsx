import React from "react";

export enum CardOffset {
	SMALL = "small",
	MIDDLE = "middle",
	LARGE = "large",
}
interface CardProps {
	topTitle?: string;
	title: string;
	offset?: CardOffset;
	children: React.ReactNode;
}

export const Card = ({ topTitle, title, offset = CardOffset.MIDDLE, children }: CardProps) => {
	const offsetValues = {
		small: "-ml-9",
		middle: "-ml-12",
		large: "-ml-14",
	};

	return (
		<div className="md:col-span-3 border border-solid rounded-[1px] border-darker w-full p-7 pt-16">
			<div className={`bg-credix-primary ${offsetValues[offset]} py-[18.5px]`}>
				<div className="text-xs font-normal capitalize">{topTitle}</div>
				<div className="font-sans font-semibold text-[32px] capitalize">{title}</div>
			</div>
			<div className="text-base">{children}</div>
		</div>
	);
};
