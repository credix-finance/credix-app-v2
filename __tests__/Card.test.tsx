import React from "react";
import renderer from "react-test-renderer";
import { Card } from "@components/Card";

const cardContent = "This is the content of the card.";

test("Card with small offset", () => {
	const component = renderer.create(
		<Card topTitle="top title" title="title" offset="small">
			{cardContent}
		</Card>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Card with middle offset", () => {
	const component = renderer.create(
		<Card topTitle="top title" title="title" offset="middle">
			{cardContent}
		</Card>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Card with large offset", () => {
	const component = renderer.create(
		<Card topTitle="top title" title="title" offset="large">
			{cardContent}
		</Card>
	);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
