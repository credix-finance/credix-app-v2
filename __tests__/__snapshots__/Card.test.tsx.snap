// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card with large offset 1`] = `
<div
  className="md:col-span-3 border border-solid rounded-[1px] border-darker w-full p-7 pt-16"
>
  <div
    className="bg-credix-primary -ml-14 py-[18.5px]"
  >
    <div
      className="text-xs font-normal capitalize"
    >
      top title
    </div>
    <div
      className="font-sans font-semibold text-[32px] capitalize"
    >
      title
    </div>
  </div>
  <div
    className="text-base"
  >
    This is the content of the card.
  </div>
</div>
`;

exports[`Card with middle offset 1`] = `
<div
  className="md:col-span-3 border border-solid rounded-[1px] border-darker w-full p-7 pt-16"
>
  <div
    className="bg-credix-primary -ml-12 py-[18.5px]"
  >
    <div
      className="text-xs font-normal capitalize"
    >
      top title
    </div>
    <div
      className="font-sans font-semibold text-[32px] capitalize"
    >
      title
    </div>
  </div>
  <div
    className="text-base"
  >
    This is the content of the card.
  </div>
</div>
`;

exports[`Card with small offset 1`] = `
<div
  className="md:col-span-3 border border-solid rounded-[1px] border-darker w-full p-7 pt-16"
>
  <div
    className="bg-credix-primary -ml-9 py-[18.5px]"
  >
    <div
      className="text-xs font-normal capitalize"
    >
      top title
    </div>
    <div
      className="font-sans font-semibold text-[32px] capitalize"
    >
      title
    </div>
  </div>
  <div
    className="text-base"
  >
    This is the content of the card.
  </div>
</div>
`;
